spec/expression.dd

```diff
@@ -2371,9 +2371,9 @@ $(GNAME TypeSpecialization):
     )
 
     $(P $(I IsExpression)s may be used in conjunction with $(D typeof) to check
-    whether an expression is semantically correct. For example, $(D is(typeof(foo)))
-    will return $(D true) if $(D foo) is a semantically correct expression with a valid
-    type.
+    whether an expression type checks correctly. For example, $(D is(typeof(foo)))
+    will return $(D true) if $(D foo) is a syntactically correct expression and if
```

Comment:
This suggests you could give it a syntactically incorrect expression and it will return `false`. In reality, you won't get past the parsing stage when you put something syntactically incorrect inside `typeof()`.

Reply:
Maybe I should just drop the "syntactically" part altogether and just state that it returns true if foo has a valid type?

Reply:
That works